master/buildbot/www/avatar.py

```diff
@@ -30,6 +32,29 @@ def getUserAvatar(self, email, size, defaultAvatarUrl):
         raise NotImplementedError()
 
 
+class AvatarGitHub(AvatarBase):
+    name = "github"
+    cache = {}
+
+    def getUserAvatar(self, email, size, defaultAvatarUrl):
+        url = self.cache.get(email)
+        if url:
+            raise resource.Redirect(url)
+
+        # https://developer.github.com/v3/search/#search-users
+        url = 'https://api.github.com/search/users?'
+        url += urllib.urlencode({'q': email, 'in': 'email', 'type': 'user'})
+        try:
+            res = json.loads(urllib.urlopen(url).read())
```

Comment:
same design pattern is use in the http reporter tests:

https://github.com/buildbot/buildbot/blob/master/master/buildbot/test/unit/test_reporter_http.py#L53